aspnetcore/mvc/models/file-uploads.md

```diff
@@ -1,195 +1,656 @@
+                name="FileUpload.FormFile">
+        </dd>
+    </dl>
+
+    <input class="btn" type="submit" value="Upload">
+
+    <div style="margin-top:15px">
+        <output name="result"></output>
     </div>
 </form>
+
+@section Scripts {
+    @{await Html.RenderPartialAsync("_ValidationScriptsPartial");}
+
+    <script>
+        "use strict";
+
+        function AJAXSubmit (oFormElement) {
+          var oReq = new XMLHttpRequest();
+          oReq.onload = function(e) { 
+            oFormElement.elements.namedItem("result").value = 
+            'Result: ' + this.status + ' ' + this.statusText;
+          };
+          oReq.open("post", oFormElement.action);
+          oReq.send(new FormData(oFormElement));
+        }
+    </script>
+}
+```
+
+The `<form>` element attribute `asp-antiforgery` is set to `true` so that the form receives an antiforgery request token. The token isn't automatically provided when a form's `action` is set explicitly. Setting the attribute to `true` overrides the default behavior. When the token is present in the form, it's automatically sent with the POST to the server to validate the request. For more information, see <xref:security/anti-request-forgery>.
+
+In order to support file uploads, HTML forms must specify an encoding type (`enctype`) of `multipart/form-data`.
+
+For a `files` input element to support uploading multiple files provide the `multiple` attribute on the `<input>` element:
+
+```cshtml
+<input asp-for="FileUpload.FormFiles" type="file" multiple>
 ```
 
-In order to support file uploads, HTML forms must specify an `enctype` of `multipart/form-data`. The `files` input element shown above supports uploading multiple files. Omit the `multiple` attribute on this input element to allow just a single file to be uploaded. The above markup renders in a browser as:
+The individual files uploaded to the server can be accessed through [Model Binding](xref:mvc/models/model-binding) using <xref:Microsoft.AspNetCore.Http.IFormFile>. The sample app demonstrates multiple buffered file uploads for database and physical storage scenarios.
+
+> [!WARNING]
+> Don't rely on or trust the `FileName` property of <xref:Microsoft.AspNetCore.Http.IFormFile> without validation. The `FileName` property should only be used for display purposes and only after HTML encoding the value.
+>
+> The examples provided don't take into account security considerations. See the [Security considerations](#security-considerations) and [Validation](#validation) sections and the sample app for information and examples of file upload validation.
+
+When uploading files using model binding and <xref:Microsoft.AspNetCore.Http.IFormFile>, the action method can accept:
+
+* A single <xref:Microsoft.AspNetCore.Http.IFormFile>.
+* An <xref:System.Collections.IEnumerable>\<<xref:Microsoft.AspNetCore.Http.IFormFile>> or [List](xref:System.Collections.Generic.List`1)\<<xref:Microsoft.AspNetCore.Http.IFormFile>> representing several files. 
```

Comment:
Also an `IFormFileCollection`

This should also call out that MVC will match form files by name when binding i.e. the `name` in `<input type="file" name="formFile">` needs to match the parameter \ property being bound just like everything else.